lib/addons/fluxcd/gitrepository.ts

```diff
@@ -0,0 +1,31 @@
+import { FluxCDAddOnProps } from '.';
+import { GitRepositoryReference } from '../../spi';
+import * as spi from "../../spi";
+
+/**
+ * Flux GitRepository API defines a Source to produce an Artifact for a Git repository revision.
+ */
+export class FluxGitRepository {
+
+    constructor(private readonly bootstrapRepo: spi.ApplicationRepository | undefined) {}
+
+    public generate(fluxcdAddonProps: FluxCDAddOnProps) {
```

Comment:
why does this function take FluxCDAddOnProps? Seems like only app repo is needed.

Reply:
Nah. We need FluxCDAddOnProps for other additional fields like namespace, interval which is not part of the app repo.